Nanoleaf-wpf/ViewModels/OptionsViewModel.cs

```diff
@@ -1,25 +1,129 @@
 ﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Winleafs.Models.Enums;
+using Winleafs.Wpf.Helpers;
+using Winleafs.Wpf.Views.Options;
 
 namespace Winleafs.Wpf.ViewModels
 {
     public class OptionsViewModel
     {
+        private OptionsWindow _parent;
+
+        #region Properties
         public bool StartAtWindowsStartUp { get; set; }
 
         public string Latitude { get; set; }
         public string Longitude { get; set; }
 
-        public int AmbilightRefreshRatePerSecond { get; set; }
-        public bool AmbilightControlBrightness { get; set; }
-
         public List<string> MonitorNames { get; set; }
 
-        public string SelectedMonitor { get; set; }
-
         public string SelectedLanguage { get; set; }
 
         public List<string> Languages { get; set; }
 
         public bool MinimizeToSystemTray { get; set; }
+        #endregion
+
+        #region Screen mirror
+        public Dictionary<string, ScreenMirrorAlgorithm> AlgorithmPerDevice { get; set; }
+        public Dictionary<string, int> ScreenMirrorRefreshRatePerDevice { get; set; }
+        public Dictionary<string, bool> ScreenMirrorControlBrightnessPerDevice { get; set; }
+        public Dictionary<string, string> MonitorPerDevice { get; set; }
+
+        private string _selectedMonitor;
+        public string SelectedMonitor
+        {
+            get => _selectedMonitor;
+            set
+            {
+                _selectedMonitor = value;
+                MonitorPerDevice[_selectedDevice] = _selectedMonitor;
+            }
+        }
+
+        private int _screenMirrorRefreshRatePerSecond;
+        public int ScreenMirrorRefreshRatePerSecond
+        {
+            get => _screenMirrorRefreshRatePerSecond;
+            set
+            {
+                _screenMirrorRefreshRatePerSecond = value;
+                ScreenMirrorRefreshRatePerDevice[_selectedDevice] = _screenMirrorRefreshRatePerSecond;
+            }
+        }
+
+        private bool _screenMirrorControlBrightness;
+        public bool ScreenMirrorControlBrightness
+        {
+            get => _screenMirrorControlBrightness;
+            set
+            {
+                _screenMirrorControlBrightness = value;
+                ScreenMirrorControlBrightnessPerDevice[_selectedDevice] = _screenMirrorControlBrightness;
+            }
+        }
+
+        #region Screen mirror algorithm dropdown
+        private Dictionary<string, ScreenMirrorAlgorithm> _screenMirrorAlgorithmMapping { get; set; } //Map display values to enum values
```

Comment:
General readability would improve if this comment is put on the line above (falls off the screen on github)